### DISCUSSION TITLE
DRILL_TO_DETAIL  Permissions to disable Edit Chart

### DISCUSSION THREAD

**Discussion by srinisubramanian** [apache/superset#25387] [Q&A / Help]
In order to display DRILL_BY context menu, Superset checks for the "explore on superset" permission.  I have a setup with a customized role so users can only view dashboards and not make any changes.  To enable DRILL_BY and DRILL_TO_DETAILI have now enabled explore on superset.  Now context menu is displayed but now user can edit chart.  In order to achieve the read only funcitonality this button needs to be disabled.

Please suggest how to configure Superset so that users have the DRILL_BY and DRILL_TO_DETAIL options (Explore on Superset enabled) but cannot make modifications / save dashboards or charts

@betodealmeida @srinify any help appreciated

**Comment by srinisubramanian** [comment]
More specifically in DRILL_TO_DETAIL I want to disable the Edit Chart button in the popup.  Any ideas?  I have customized roles so changing permissions is ok but not sure which permission I should tweak

<img width="601" alt="image" src="https://github.com/apache/superset/assets/8343179/8ba8b74d-d30b-432d-95f8-b1dde4f75b03">

**Reply by duniang818** [reply]
@srinisubramanian i have same need with you, do you have solved ?

**Reply by srinisubramanian** [reply]
Nope. For now living  with error that pops up for users that access is denied when they click on the button.

**Comment by nagarajmmu** [comment]
We are also facing the same issue, if anybody resolved this issue, please let us know. it would be a great help

**Comment by cmcnicoll** [comment]
I'm also interested in a separate permission to disable the Edit Chart button.

**Comment by rscarborough1996** [comment]
Permissions are changing in this recent pull request: #27029 

The author says in a comment that, while `can explore` isn't necessary for drill by, the permissions should be separated in a follow up PR.
